examples/simple-source/src/main.rs

```diff
@@ -48,6 +52,10 @@ pub(crate) mod simple_source {
                     return;
                 }
 
+                let mut headers = HashMap::new();
+                headers.insert(String::from("x-txn-id"), String::from(Uuid::new_v4()));
+                let shared_headers = Arc::new(headers);
+
```

Comment:
Why does this need to be an Arc? Since we create a new set of headers for every message, can we just give the ownership?

Reply:
I guess then we have to remove this https://github.com/numaproj/numaflow-rs/blob/main/src/source.rs#L200 from here

Reply:
please remove if it is not required